x-pack/plugins/fleet/server/services/epm/package_service.ts

```diff
@@ -86,8 +87,17 @@ export class PackageServiceImpl implements PackageService {
   ) {}
 
   public asScoped(request: KibanaRequest) {
-    const preflightCheck = () => {
-      if (!checkSuperuser(request)) {
+    const preflightCheck = async (requiredAuthz?: FleetAuthzRouteConfig['fleetAuthz']) => {
+      if (requiredAuthz) {
+        const requestedAuthz = await getAuthzFromRequest(request);
+
+        const noRequiredAuthz = doesNotHaveRequiredFleetAuthz(requestedAuthz, requiredAuthz);
+        if (noRequiredAuthz === undefined || noRequiredAuthz === true) {
+          throw new FleetUnauthorizedError(
+            `User does not have adequate permissions to access Fleet packages.`
+          );
+        }
+      } else if (!checkSuperuser(request)) {
```

Comment:
TODO: superuser check to be replaced by specific privileges for each service method